### PR TITLE
Add changelog of EMQX e5.4.1

### PR DESCRIPTION
**Pull request by yanzhiemq**

### CODE REVIEW

File: zh_CN/changes/changes-ee-v5.md [hunk 1]
```diff
@@ -1,5 +1,20 @@
 # v5 版本
 
+## 5.4.1
+
+*发布日期: 2024-01-05*
+
+### 增强
+
+- [#12234](https://github.com/emqx/emqx/pull/12234) 解决了 EMQX 5.4.0 之前版本在 `emqx.conf` 中定义的 Open Telemetry 配置的兼容性问题，确保最新 EMQX 发布版本能够平滑兼容旧版配置。
+- [#12236](https://github.com/emqx/emqx/pull/12236) 实现了使用简短的 MQTT Sink/Source 客户端 ID。
```
Comment (wivwiv) [inline zh_CN/changes/changes-ee-v5.md]:
```suggestion
- [#12236](https://github.com/emqx/emqx/pull/12236) MQTT Sink/Source 使用简短的客户端 ID。
```

File: zh_CN/changes/changes-ee-v5.md [hunk 2]
```diff
@@ -1,5 +1,20 @@
...
+- [#12238](https://github.com/emqx/emqx/pull/12238) 增强了与 EMQX 5.3.2 版本中 HTTP Action 功能引入的错误格式配置的兼容性。
+- [#12240](https://github.com/emqx/emqx/pull/12240) 修改了 `/file_transfer` API，使其以原始格式返回文件传输配置。此更改防止了时间单位（如 "1h"）转换为秒，确保调用者接收到初始配置的值。此修改与其他 getter API 保持了数据表示的一致性。
```
Comment (wivwiv) [inline zh_CN/changes/changes-ee-v5.md]:
```suggestion
- [#12240](https://github.com/emqx/emqx/pull/12240) 修改了 `/file_transfer` REST API，按照原始格式返回配置，避免将时间单位（如 "1h"）转换为秒，确保调用者接收到初始配置的值，此修改与其他 GET API 保持一致的数据格式。
```

File: zh_CN/changes/changes-ee-v5.md [hunk 3]
```diff
@@ -1,5 +1,20 @@
...
+- [#12241](https://github.com/emqx/emqx/pull/12241) 修复了配置额外的 S3 API HTTP 头导致使用 S3 存储后端的文件传输中断的问题，确保稳定且不间断的文件传输操作。
```
Comment (wivwiv) [inline zh_CN/changes/changes-ee-v5.md]:
```suggestion
- [#12241](https://github.com/emqx/emqx/pull/12241) 修复了配置额外的 S3 HTTP 请求头导致文件传输中断的问题，确保稳定且不间断的文件传输操作。
```

File: zh_CN/changes/changes-ee-v5.md [hunk 4]
```diff
@@ -1,5 +1,20 @@
...
+- [#12246](https://github.com/emqx/emqx/pull/12246) 停止在 Docker 中默认暴露 11883 端口，并从 Helm 图表中移除，因为此端口不再使用。
```
Comment (wivwiv) [inline zh_CN/changes/changes-ee-v5.md]: 专有名词
```suggestion
- [#12246](https://github.com/emqx/emqx/pull/12246) 停止在 Docker 中默认暴露不再使用的 11883 端口，并从 Helm  Chart 中移除。
```

**Review by wivwiv — CHANGES_REQUESTED** [review]

File: en_US/changes/changes-ee-v5.md [hunk 5]
```diff
@@ -1,5 +1,29 @@
...
 
+## 5.4.1
+
+*Release Date: 2024-01-05*
+
+### Bug Fixes
+
+- [#12234](https://github.com/emqx/emqx/pull/12234) Resolved compatibility issues with Open Telemetry configurations defined in `emqx.conf` from versions before EMQX 5.4.0, ensuring smooth integration of legacy configurations with the latest EMQX release.
+
+- [#12236](https://github.com/emqx/emqx/pull/12236) Implemented the use of short client IDs for MQTT Sink/Source.
```
Comment (id) [inline en_US/changes/changes-ee-v5.md]: This sound like an enhancement?

File: en_US/changes/changes-ee-v5.md [hunk 6]
```diff
@@ -1,5 +1,29 @@
...
+
+- [#12238](https://github.com/emqx/emqx/pull/12238) Enhanced compatibility with the error format configurations introduced in the HTTP Action feature of EMQX version 5.3.2.
```
Comment (id) [inline en_US/changes/changes-ee-v5.md]: This too sounds like an enhancement.

File: en_US/changes/changes-ce-5.4.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
+# Releases
+
+## e5.4.0
```
Comment (id) [inline en_US/changes/changes-ce-5.4.md]:
```suggestion
## v5.4.0
```

File: en_US/changes/changes-ce-5.4.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
+# Releases
```
Comment (id) [inline en_US/changes/changes-ce-5.4.md:1]: Shouldn't we add 5.4.1 section here as well?

File: zh_CN/changes/changes-ce-5.4.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
+# 版本发布
+
+## e5.4.0
```
Comment (id) [inline zh_CN/changes/changes-ce-5.4.md]:
```suggestion
## v5.4.0
```

File: en_US/changes/changes-ee-v5.md (hunk 5 above)
Comment (yanzhiemq) [reply en_US/changes/changes-ee-v5.md]: This PR is used to fix the incompatible bug introduced in e5.3.2

File: en_US/changes/changes-ee-v5.md (hunk 6 above)
Comment (yanzhiemq) [reply en_US/changes/changes-ee-v5.md]: This PR is used to fix a bug reported from our customer that the client id is too long to be integrated with their system.

File: en_US/changes/changes-ce-5.4.md (hunk 8 above)
Comment (Meggielqk) [reply en_US/changes/changes-ce-5.4.md:1]: This file and the correspondence ee file will not be used anymore. We have put all the previous release history in one file, as required by CEO.

File: zh_CN/changes/changes-ee-v5.md [hunk 10]
```diff
@@ -1,5 +1,21 @@
 # v5 版本
 
+## 5.4.1
+
+*发布日期: 2024-01-05*
+
+### 增强
+
+- [#12234](https://github.com/emqx/emqx/pull/12234) 解决了 EMQX 5.4.0 之前版本在 `emqx.conf` 中定义的 Open Telemetry 配置的兼容性问题，确保最新 EMQX 发布版本能够平滑兼容旧版配置。
+- [#12236](https://github.com/emqx/emqx/pull/12236) MQTT Sink/Source 使用简短的客户端 ID。
```
Comment (yanzhiemq) [inline zh_CN/changes/changes-ee-v5.md]:
```suggestion
- [#12236](https://github.com/emqx/emqx/pull/12236) 修复了 MQTT Sink/Source 客户端 ID 生成方法，以符合 MQTT 3.1 规范的 23 字节限制。 客户端 ID 现在以用户分配的数据集成名称为前缀，后跟节点名称的 SHA 哈希值和池成员 ID 的前 8 个字节。 如果生成的 ID 超过 23 字节，则会应用额外的 SHA 哈希和截断以确保合规性。
```

File: zh_CN/changes/changes-ee-v5.md [hunk 11]
```diff
@@ -1,5 +1,21 @@
...
+- [#12238](https://github.com/emqx/emqx/pull/12238) 增强了与 EMQX 5.3.2 版本中 HTTP Action 功能引入的错误格式配置的兼容性。
```
Comment (yanzhiemq) [inline zh_CN/changes/changes-ee-v5.md]:
```suggestion
- [#12238](https://github.com/emqx/emqx/pull/12238) 解决了EMQX 5.3.2 版本 HTTP Action 功能中引入的错误格式配置的兼容性问题。
```

File: zh_CN/changes/changes-ee-v5.md [hunk 12]
```diff
@@ -1,5 +1,21 @@
 # v5 版本
 
+## 5.4.1
+
+*发布日期: 2024-01-05*
+
+### 增强
```
Comment (yanzhiemq) [inline zh_CN/changes/changes-ee-v5.md]:
```suggestion
### 修复
```

**Review by wivwiv — DISMISSED** [review]

File: en_US/changes/changes-ee-v5.md [hunk 13]
```diff
@@ -1,5 +1,28 @@
...
+
+- [#12241](https://github.com/emqx/emqx/pull/12241) Fixed a bug where configuring additional HTTP headers for S3 API interactions disrupted file transfers using the S3 storage backend, ensuring stable and uninterrupted file transfer operations.
+
+- [#12246](https://github.com/emqx/emqx/pull/12246) Stopped exposing port 11883 by default in Docker and removed it from Helm charts, as this port is no longer in use. 
+
+- [#12249](https://github.com/emqx/emqx/pull/12249) Fixed an issue in the `/configs` API where attempting to modify a read-only configuration value resulted in a garbled response message.
+
+- [#12250](https://github.com/emqx/emqx/pull/12250) Resolved an issue where the `file_transfer` configuration's `secret_access_key` value was erroneously being updated to masked stars (`*****`), ensuring that the original key value remains unaltered and secure.
+- [#12256](https://github.com/emqx/emqx/pull/12256) Fixed an issue that prevented establishing connections to MySQL resources without a password.
+
```
Comment (yanzhiemq) [inline en_US/changes/changes-ee-v5.md:24]:
```suggestion

- [#12264](https://github.com/emqx/emqx/pull/12264) Fix rolling upgrade when replica nodes join cluster with core nodes prior to version 5.4.
```

File: en_US/changes/changes-ce-v5.md [hunk 14]
```diff
@@ -1,8 +1,18 @@
...
+- [#12236](https://github.com/emqx/emqx/pull/12236) Fixed client ID generation in MQTT broker data integration to comply with MQTT 3.1 specification of 23-byte limit. Client ID is now prefixed with user-assigned Connector name, followed by the first 8 bytes of node name's SHA hash and pool member ID. If the resulting ID exceeds 23 bytes, additional SHA hash and truncation for the first 23 characters are applied to ensure compliance.
+- [#12238](https://github.com/emqx/emqx/pull/12238) Resolved compatibility issue with the error format configurations introduced in the HTTP Action feature of EMQX version 5.3.2.
+- [#12246](https://github.com/emqx/emqx/pull/12246) Stopped exposing port 11883 by default in Docker and removed it from Helm charts, as this port is no longer in use. 
+- [#12249](https://github.com/emqx/emqx/pull/12249) Fixed an issue in the `/configs` API where attempting to modify a read-only configuration value resulted in a garbled response message.
+
```
Comment (yanzhiemq) [inline en_US/changes/changes-ce-v5.md:17]:
```suggestion
- [#12264](https://github.com/emqx/emqx/pull/12264) Fix rolling upgrade when replica nodes join cluster with core nodes prior to version 5.4.

```

File: zh_CN/changes/changes-ee-v5.md [hunk 15]
```diff
@@ -1,5 +1,21 @@
...
+
+- [#12234](https://github.com/emqx/emqx/pull/12234) 解决了 EMQX 5.4.0 之前版本在 `emqx.conf` 中定义的 Open Telemetry 配置的兼容性问题，确保最新 EMQX 发布版本能够平滑兼容旧版配置。
+- [#12236](https://github.com/emqx/emqx/pull/12236) 修复了 MQTT 服务数据集成中客户端 ID 的生成方法，以符合 MQTT 3.1 规范的 23 字节限制。 客户端 ID 现在以用户分配的连接器名称为前缀，后跟节点名称的 SHA 哈希值和池成员 ID 的前 8 个字节。 如果生成的 ID 超过 23 字节，则会将 ID 重新使用 SHA 哈希，并取哈希的前 23 个字符以确保合规性。
+- [#12238](https://github.com/emqx/emqx/pull/12238) 解决了EMQX 5.3.2 版本 HTTP Action 功能中引入的错误格式配置的兼容性问题。
+- [#12240](https://github.com/emqx/emqx/pull/12240) 修改了 `/file_transfer` REST API，按照原始格式返回配置，避免将时间单位（如 "1h"）转换为秒，确保调用者接收到初始配置的值，此修改与其他 GET API 保持一致的数据格式。
+- [#12241](https://github.com/emqx/emqx/pull/12241) 修复了配置额外的 S3 HTTP 请求头导致文件传输中断的问题，确保稳定且不间断的文件传输操作。
+- [#12246](https://github.com/emqx/emqx/pull/12246) 停止在 Docker 中默认暴露不再使用的 11883 端口，并从 Helm  Chart 中移除。
+- [#12249](https://github.com/emqx/emqx/pull/12249) 修复了 `/configs` API 中尝试修改只读配置值导致响应消息乱码的问题。
+- [#12250](https://github.com/emqx/emqx/pull/12250) 解决了 `file_transfer` 配置的 `secret_access_key` 值错误更新为掩码星号 (`*****`) 的问题，确保原始密钥值保持不变以保证安全性。
+- [#12256](https://github.com/emqx/emqx/pull/12256) 修复了没有密码就无法与 MySQL 资源建立连接的问题。
```
Comment (yanzhiemq) [inline zh_CN/changes/changes-ee-v5.md:17]:
```suggestion
- [#12256](https://github.com/emqx/emqx/pull/12256) 修复了没有密码就无法与 MySQL 资源建立连接的问题。
- [#12264](https://github.com/emqx/emqx/pull/12264) 修复了滚动升级时5.4的副本节点加入5.4之前的核心节点失败问题。
```

File: zh_CN/changes/changes-ce-v5.md [hunk 16]
```diff
@@ -1,5 +1,17 @@
...
+
+*发布日期: 2024-01-05*
+
+### 修复
+
+- [#12234](https://github.com/emqx/emqx/pull/12234) 解决了 EMQX 5.4.0 之前版本在 `emqx.conf` 中定义的 Open Telemetry 配置的兼容性问题，确保最新 EMQX 发布版本能够平滑兼容旧版配置。
+- [#12236](https://github.com/emqx/emqx/pull/12236) 修复了 MQTT 服务数据集成中客户端 ID 的生成方法，以符合 MQTT 3.1 规范的 23 字节限制。 客户端 ID 现在以用户分配的连接器名称为前缀，后跟节点名称的 SHA 哈希值和池成员 ID 的前 8 个字节。 如果生成的 ID 超过 23 字节，则会将 ID 重新使用 SHA 哈希，并取哈希的前 23 个字符以确保合规性。
+- [#12238](https://github.com/emqx/emqx/pull/12238) 解决了EMQX 5.3.2 版本 HTTP Action 功能中引入的错误格式配置的兼容性问题。
+- [#12246](https://github.com/emqx/emqx/pull/12246) 停止在 Docker 中默认暴露不再使用的 11883 端口，并从 Helm  Chart 中移除。
+- [#12249](https://github.com/emqx/emqx/pull/12249) 修复了 `/configs` API 中尝试修改只读配置值导致响应消息乱码的问题。
```
Comment (yanzhiemq) [inline zh_CN/changes/changes-ce-v5.md:14]:
```suggestion
- [#12249](https://github.com/emqx/emqx/pull/12249) 修复了 `/configs` API 中尝试修改只读配置值导致响应消息乱码的问题。
- [#12264](https://github.com/emqx/emqx/pull/12264) 修复了滚动升级时5.4的副本节点加入5.4之前的核心节点失败问题。
```

File: en_US/changes/changes-ce-v5.md [hunk 17]
```diff
@@ -1,8 +1,19 @@
 # Version 5
 
-## 5.4.0
+## 5.4.1
+
+*Release Date: 2024-01-05*
```
Comment (yanzhiemq) [inline en_US/changes/changes-ce-v5.md]:
```suggestion
*Release Date: 2024-01-09*

```

File: en_US/changes/changes-ee-v5.md [hunk 18]
```diff
@@ -1,5 +1,29 @@
 # Version 5
 
+## 5.4.1
+
+*Release Date: 2024-01-05*
```
Comment (yanzhiemq) [inline en_US/changes/changes-ee-v5.md]:
```suggestion
*Release Date: 2024-01-09*
```

File: zh_CN/changes/changes-ce-v5.md [hunk 19]
```diff
@@ -1,5 +1,18 @@
 # v5 版本
 
+## 5.4.1
+
+*发布日期: 2024-01-05*
```
Comment (yanzhiemq) [inline zh_CN/changes/changes-ce-v5.md]:
```suggestion
*发布日期: 2024-01-09*

```

File: zh_CN/changes/changes-ee-v5.md [hunk 20]
```diff
@@ -1,5 +1,22 @@
 # v5 版本
 
+## 5.4.1
+
+*发布日期: 2024-01-05*
```
Comment (yanzhiemq) [inline zh_CN/changes/changes-ee-v5.md]:
```suggestion
*发布日期: 2024-01-09*
```

**Review by Meggielqk — DISMISSED** [review]

File: en_US/changes/changes-ce-v5.md [hunk 21]
```diff
@@ -1,8 +1,20 @@
...
+- [#12236](https://github.com/emqx/emqx/pull/12236) Fixed client ID generation in MQTT broker data integration to comply with MQTT 3.1 specification of 23-byte limit. Client ID is now prefixed with user-assigned Connector name, followed by the first 8 bytes of node name's SHA hash and pool member ID. If the resulting ID exceeds 23 bytes, additional SHA hash and truncation for the first 23 characters are applied to ensure compliance.
+- [#12238](https://github.com/emqx/emqx/pull/12238) Resolved compatibility issue with the error format configurations introduced in the HTTP Action feature of EMQX version 5.3.2.
+- [#12246](https://github.com/emqx/emqx/pull/12246) Stopped exposing port 11883 by default in Docker and removed it from Helm charts, as this port is no longer in use. 
+- [#12249](https://github.com/emqx/emqx/pull/12249) Fixed an issue in the `/configs` API where attempting to modify a read-only configuration value resulted in a garbled response message.
+- [#12264](https://github.com/emqx/emqx/pull/12264) Fixed rolling upgrade when replica nodes join cluster with core nodes prior to version 5.4.
```
Comment (yanzhiemq) [inline en_US/changes/changes-ce-v5.md]:
```suggestion
- [#12264](https://github.com/emqx/emqx/pull/12264) Fixed an issue where version 5.4 replica nodes could not join clusters with core nodes running versions earlier than 5.4 during the rolling upgrade process.

```

File: en_US/changes/changes-ee-v5.md [hunk 22]
```diff
@@ -1,5 +1,29 @@
...
+- [#12241](https://github.com/emqx/emqx/pull/12241) Fixed a bug where configuring additional HTTP headers for S3 API interactions disrupted file transfers using the S3 storage backend, ensuring stable and uninterrupted file transfer operations.
+
+- [#12246](https://github.com/emqx/emqx/pull/12246) Stopped exposing port 11883 by default in Docker and removed it from Helm charts, as this port is no longer in use. 
+
+- [#12249](https://github.com/emqx/emqx/pull/12249) Fixed an issue in the `/configs` API where attempting to modify a read-only configuration value resulted in a garbled response message.
+
+- [#12250](https://github.com/emqx/emqx/pull/12250) Resolved an issue where the `file_transfer` configuration's `secret_access_key` value was erroneously being updated to masked stars (`*****`), ensuring that the original key value remains unaltered and secure.
+- [#12256](https://github.com/emqx/emqx/pull/12256) Fixed an issue that prevented establishing connections to MySQL resources without a password.
+
+- [#12264](https://github.com/emqx/emqx/pull/12264) Fixed rolling upgrade when replica nodes join cluster with core nodes prior to version 5.4.
```
Comment (yanzhiemq) [inline en_US/changes/changes-ee-v5.md]:
```suggestion
- [#12264](https://github.com/emqx/emqx/pull/12264) Fixed an issue where version 5.4 replica nodes could not join clusters with core nodes running versions earlier than 5.4 during the rolling upgrade process.```

File: en_US/changes/changes-ee-v5.md [hunk 23]
```diff
@@ -1,5 +1,29 @@
...
+
+- [#12246](https://github.com/emqx/emqx/pull/12246) Stopped exposing port 11883 by default in Docker and removed it from Helm charts, as this port is no longer in use. 
+
+- [#12249](https://github.com/emqx/emqx/pull/12249) Fixed an issue in the `/configs` API where attempting to modify a read-only configuration value resulted in a garbled response message.
+
+- [#12250](https://github.com/emqx/emqx/pull/12250) Resolved an issue where the `file_transfer` configuration's `secret_access_key` value was erroneously being updated to masked stars (`*****`), ensuring that the original key value remains unaltered and secure.
+- [#12256](https://github.com/emqx/emqx/pull/12256) Fixed an issue that prevented establishing connections to MySQL resources without a password.
+
+- [#12264](https://github.com/emqx/emqx/pull/12264) Fixed an issue where version 5.4 replica nodes could not join clusters with core nodes running versions earlier than 5.4 during the rolling upgrade process.```
```
Comment (yanzhiemq) [inline en_US/changes/changes-ee-v5.md]:
```suggestion
- [#12264](https://github.com/emqx/emqx/pull/12264) Fixed an issue where version 5.4 replica nodes could not join clusters with core nodes running versions earlier than 5.4 during the rolling upgrade process.
```

**Review by wivwiv — APPROVED** [review]

**Pull request merged**